### PR TITLE
Fix lost currency on recurring expense form

### PR DESCRIPTION
**Pull request by Betree**
Fix https://github.com/opencollective/opencollective/issues/6379

TLDR; we no longer consider the form "ready" if a user is logged in and we haven't yet reloaded the data for him.

This happened because the 1st fetch, using the `draftKey` while not authenticated yet, did not return any `payoutMethod` because of permissions. I added a condition not to reset the currency in the ExpenseForm when `loading` is true, but there was then an intermediate state between the two that got the form to reset. It basically looked like that:

| Step | LoggedInUser | LoadingLoggedInUser | RefetchingDataForUser | Payout |
| ----- | ---- | ---- |---- | ---- |
| 1 | :x: | :heavy_check_mark: | :x: | :x: |
| 2 | :heavy_check_mark: | :x: | :x: | :x: |
| 3 | :heavy_check_mark: | :x: | :heavy_check_mark: | :x: |
| 4 | :heavy_check_mark: | :x: | :x: | :heavy_check_mark: |

This 2nd step was the problematic one, and it was happening because we set `RefetchingDataForUser` in a `componentDidUpdate`, which happens after the render. I changed the strategy to record a **hasRefetched**DataForUser rather than (**isRefetching**DataForUser). In `render`, `isRefetchingDataForUser` is now defined as: `LoggedInUser && !hasRefetchedDataForUser` which produce the following renders:

| Step | LoggedInUser | LoadingLoggedInUser | RefetchingDataForUser | Payout |
| ----- | ---- | ---- |---- | ---- |
| 1 | :x: | :heavy_check_mark: | :x: | :x: |
| 2 | :heavy_check_mark: | :x: | :heavy_check_mark: | :x: |
| 3 | :heavy_check_mark: | :x: | :heavy_check_mark: | :x: |
| 4 | :heavy_check_mark: | :x: | :x: | :heavy_check_mark: |

This will fix the root issue and prevent similar ones where we automatically update values based on `data`.

### CODE REVIEW

File: components/expenses/ExpenseForm.js [hunk 1]
```diff
@@ -362,7 +362,7 @@ const ExpenseFormBody = ({
         formik.setFieldValue('currency', null);
       }
     }
-  }, [values.payoutMethod]);
+  }, [loading, values.payoutMethod]);
```
Comment (kewitz) [inline components/expenses/ExpenseForm.js:365]: 👍🏼

**Review by kewitz — APPROVED** [review]

**Comment by mkurz** [comment]
Does happen again (or never was fixed).
